### PR TITLE
Add path request send maximum

### PR DESCRIPTION
**Pull request by miguelportilla**
Reviewers: @JoelKatz @ximinez

### CODE REVIEW

File: src/ripple/app/paths/PathRequest.cpp [hunk 1]
```diff
@@ -175,8 +174,30 @@ void PathRequest::updateComplete ()
...
+            }
+        }
+        else
+        {
+            if (saDstAmount <= zero && saDstAmount != sa)
+            {
+                bValid = false;
+                jvStatus = rpcError(rpcDST_AMT_MALFORMED);
+            }
+        }
```
Comment (ximinez) [inline src/ripple/app/paths/PathRequest.cpp]: These nested `if` statements could be flattened a bit.
```
        if (saSendMax && saDstAmount != sa)
        {
            // If send max specified, dst amt must be -1
            bValid = false;
            jvStatus = rpcError(rpcDST_AMT_MALFORMED);
        }
        else if (!saSendMax && saDstAmount <= zero && saDstAmount != sa)
        {
            bValid = false;
            jvStatus = rpcError(rpcDST_AMT_MALFORMED);
        }
```

File: src/ripple/app/paths/PathRequest.cpp [hunk 2]
```diff
@@ -408,6 +478,87 @@ void PathRequest::resetLevel (int l)
...
+
+        jvArray.append (jvEntry);
+        return true;
+    }
+    
+    m_journal.debug << iIdentifier << " rippleCalc returns "
+        << transHuman (rc.result ());
+    return false;
+}
```
Comment (ximinez) [inline src/ripple/app/paths/PathRequest.cpp]: While I don't completely understand this function entirely yet, I did want to note for myself and others that I have verified that if faithfully migrates the functionality of the removed block starting at https://github.com/ripple/rippled/pull/1209/files#diff-39a6cdaf81297515fe5f026a03ec2addL515

File: src/ripple/app/paths/PathRequest.cpp [hunk 3]
```diff
@@ -380,7 +424,33 @@ int PathRequest::parseJson (Json::Value const& jvParams, bool complete)
                 uIss = *raSrcAccount;
             }
 
-            sciSourceCurrencies.insert ({uCur, uIss});
+            if (raSrcAccount && saSendMax)
```
Comment (ximinez) [inline src/ripple/app/paths/PathRequest.cpp]: Does this mean that send_max is basically ignored if there is no source_account? Should that case be a parsing error?

Comment (miguelportilla) [inline src/ripple/app/paths/PathRequest.cpp]: They way parsejson is called, the source account can never be null at this point.

I didn't want to disturb the code more than I needed to but I think I'll get rid of the unnecessary complete param as its the cause of some ugliness. The method is private and only called with it being true from a single place.

Comment (ximinez) [inline src/ripple/app/paths/PathRequest.cpp]: Ok, that makes sense.

**Comment by JoelKatz** [comment]
I can't seem to get it to actually work. Tried with:
{
    "source_account" : "rMz5b4HYedtiKcyn18B2a6EL9pGasKavqZ",
    "destination_account" : "rwcT5BaDjPoPyvTBig2GqHnopxbBeWG4yX",
    "send_max" : "126283730",
    "destination_amount" :
     { "currency" : "USD", "value" : "-1", "issuer" : "rwcT5BaDjPoPyvTBig2GqHnopxbBeWG4yX" }
}
Get:
      "error" : "dstAmtMalformed",
      "error_code" : 44,
      "error_message" : "Destination amount/currency/issuer is malformed.",

**Comment by JoelKatz** [comment]
Seems to be this check in RipplePathFind.cpp, `|| saDstAmount <= zero`. This allows the request to execute but I never get a sensible reply from pathfinding.

**Pull request closed without merging**